service/history/engine/engineimpl/register_domain_failover_callback.go

```diff
@@ -204,14 +215,16 @@ func (e *historyEngineImpl) generateGracefulFailoverTasksForDomainUpdateCallback
 
 func (e *historyEngineImpl) lockTaskProcessingForDomainUpdate() {
 	e.logger.Debug("Locking processing for domain update")
-	e.txProcessor.LockTaskProcessing()
-	e.timerProcessor.LockTaskProcessing()
+	for _, processor := range e.queueProcessors {
```

Comment:
not in the scope of this PR but let's see if we really need to lock task processing for these domain change callbacks.